src/NRediSearch/InfoResult.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System.Collections.Generic;
+using StackExchange.Redis;
+
+namespace NRediSearch
+{
+    public class InfoResult
+    {
+        public string IndexName { get; }
+
+        public Dictionary<string, RedisResult[]> Fields { get; } = new Dictionary<string, RedisResult[]>();
+
+        public long NumDocs { get; }
+
+        public long NumTerms { get; }
+
+        public long NumRecords { get; }
+
+        public double InvertedSzMb { get; }
+
+        public double InvertedCapMb { get; }
+
+        public double InvertedCapOvh { get; }
+
+        public double OffsetVectorsSzMb { get; }
+
+        public double SkipIndexSizeMb { get; }
+
+        public double ScoreIndexSizeMb { get; }
+
+        public double RecordsPerDocAvg { get; }
+
+        public double BytesPerRecordAvg { get; }
+
+        public double OffsetsPerTermAvg { get; }
+
+        public double OffsetBitsPerRecordAvg { get; }
+
+        public string MaxDocId { get; }
+
+        public double DocTableSizeMb { get; }
+
+        public double SortableValueSizeMb { get; }
+
+        public double KeyTableSizeMb { get; }
+
+        public Dictionary<string, RedisResult> GcStats { get; } = new Dictionary<string, RedisResult>();
+
+        public Dictionary<string, RedisResult> CursorStats { get; } = new Dictionary<string, RedisResult>();
+
+        public InfoResult(RedisResult result)
+        {
+            var results = (RedisResult[])result;
+
+            for (var i = 0; i < results.Length; i += 2)
```

Comment:
do we actually need to parse these all out? most of these feel like things that we could lookup on-the-fly cheaply, i.e.

``` c#
public double DocTableSizeMb => GetDouble("doc_table_size_mb");
private double GetDouble(string key)
    => all.TryGetValue(key, out var value) ? (double)value : default;
```

Reply:
That seems like it's be a lot less gross.